### DISCUSSION TITLE
Insert числовой переменной в переменную select другой таблицы

### DISCUSSION THREAD

**Discussion by Grume** [totumonline/totum-ru-issues-and-discussions#102] [Как это сделать?] [answered]
Пытаюсь вставить переменную #nomer с числовым значением номера строки из исходной таблицы в другую таблицу с переменной с типом данных селект, но функция инсерт вставляет именно числовое значение в поле селект. Как выбрать именно селектовое значение?
a1=: insert(table: 'klienty'; field: 'nazvanie' = $#nfv; field: 'email' = #pochta; **field: 'svyazannyy_lid' = #nomer**; inserts: "inserts"; log: true)

Также второй вопрос: если данные из одной таблицы переносятся в другую, то правильно их переносить из кода каждого столбца или в коде одного столбца задействовать все переменные из таблицы в коде? Если первое, то как указать каждой функции инсерт в столбцах, что информацию нужно вносить именно в одну строку, а не каждый раз создавать новую строку?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Вам нужно еще раз посмотреть все про Селекты. Селект это Значение с привязанным к нему Отображением. Когда вы чтото делаете `set` или `insert` или рассчитываете кодом в поле Селект вы работаете со Значением.

Если у вас в `svyazannyy_lid` селект стандартный, то он привязан к `id` таблицы источника. Отсюда возможно что ваш `nomer` с `id` не совпадает. И я бы в этом случае рекомендовал передавать именно `id`.

Если данные одной таблицы дублируются в другую, о правильно сделать один код-действия при добавлении (обычно в первом поле делается, чтоб потом не искать). Речь идет именно про дублирование при добавлении. Если вам после добавления нужно обеспечить синхрон — то самый простой способ это код-действия на изменение с `set` в нужных полях. Есть более продвинутые способы через агрегируемое поле в конце строки, но это после того как вы базовый механизм освоите.

Самым хорошим способом при этом является передавать все значения в ведущее поле целевой таблицы и оттуда растаскивать кодами. Вот здесь это показано для автозаполнения, но такая же логика работает и для `insert`: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/33
